hooks/hook.go

```diff
@@ -0,0 +1,37 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+// The hooks package contains all hook runners supported by snapd.
+package hooks
+
+import (
+	"gopkg.in/tomb.v2"
+)
+
+// HookRef is a reference to a hook within a specific snap.
+type HookRef struct {
+	Snap string `json:"snap"`
```

Comment:
the use cases I could think of would be of the sort asking previous revisions to release something,  it may be that we would need that for the current active one and there wouldn't be a new active one until that is done, anyway in general so far tasks tries to be precise about revisions so from that point of view carrying a revision would make sense even it it's unused for a while